Research/QuantBook.cs

```diff
@@ -144,62 +172,144 @@ public QuantBook() : base()
         }
 
         /// <summary>
-        /// Get fundamental data from given symbols
+        /// Python implementation of GetFundamental, get fundamental data for input symbols or tickers
         /// </summary>
-        /// <param name="pyObject">The symbols to retrieve fundamental data for</param>
+        /// <param name="input">The symbols or tickers to retrieve fundamental data for</param>
         /// <param name="selector">Selects a value from the Fundamental data to filter the request output</param>
         /// <param name="start">The start date of selected data</param>
         /// <param name="end">The end date of selected data</param>
-        /// <returns></returns>
-        public PyObject GetFundamental(PyObject tickers, string selector, DateTime? start = null, DateTime? end = null)
+        /// <returns>pandas DataFrame</returns>
+        public PyObject GetFundamental(PyObject input, string selector, DateTime? start = null, DateTime? end = null)
         {
+            //Null selector is not allowed for Python DataFrame
             if (string.IsNullOrWhiteSpace(selector))
             {
-                return "Invalid selector. Cannot be None, empty or consist only of white-space characters".ToPython();
+                throw new Exception("Invalid selector. Cannot be None, empty or consist only of white-space characters");
```

Comment:
Think would be better to use `ArgumentException` here instead, same for the ones at `PyConvertToSymbols`